load-testing/tests/relays_stress_helpers_test.go

```diff
@@ -0,0 +1,872 @@
+
+	return &accountInfo{
+		accAddress:    accAddress,
+		keyName:       keyName,
+		pendingMsgs:   []sdk.Msg{},
+		amountToStake: amountToStake,
+	}
+}
+
+// getAppFundingAmount calculates the application funding amount based on the
+// remaining test duration in blocks, the relay rate per application, the relay
+// cost, and the block duration.
+func (s *relaysSuite) getAppFundingAmount(currentBlockHeight int64) sdk.Coin {
+	currentTestDuration := s.startBlockHeight + s.testDurationBlocks - currentBlockHeight
+	// Multiply by 2 to make sure the application does not run out of funds.
+	appFundingAmount := s.relayRatePerApp * s.relayCost * currentTestDuration * blockDuration * 2
+	return sdk.NewCoin("upokt", math.NewInt(appFundingAmount))
+}
+
+// generateFundApplicationMsg generates a bank.MsgSend message to fund a given
+// application and appends it to the funding account's pending messages.
+// No transaction is sent to give flexibility to the caller to group multiple
+// messages in a single transaction.
+func (s *relaysSuite) generateFundApplicationMsg(application *accountInfo) {
+	fundAppMsg := banktypes.NewMsgSend(
+		s.fundingAccountInfo.accAddress,
+		application.accAddress,
+		sdk.NewCoins(application.amountToStake),
+	)
+
+	s.fundingAccountInfo.pendingMsgs = append(s.fundingAccountInfo.pendingMsgs, fundAppMsg)
+}
+
+// generateStakeApplicationMsg generates a MsgStakeApplication message to stake a given
+// application then appends it to the application account's pending messages.
+// No transaction is sent to give flexibility to the caller to group multiple
+// application messages into a single transaction which is useful for staking
+// then delegating to multiple gateways in the same transaction.
+func (s *relaysSuite) generateStakeApplicationMsg(application *accountInfo) {
+	stakeAppMsg := apptypes.NewMsgStakeApplication(
+		application.accAddress.String(),
+		application.amountToStake,
+		[]*sharedtypes.ApplicationServiceConfig{{Service: usedService}},
+	)
+
+	application.pendingMsgs = append(application.pendingMsgs, stakeAppMsg)
+}
+
+// generateDelegateToGatewayMsg generates a MsgDelegateToGateway message to delegate
+// a given application to a given gateway then appends it to the application account's
+// pending messages.
+func (s *relaysSuite) generateDelegateToGatewayMsg(application, gateway *accountInfo) {
+	delegateMsg := apptypes.NewMsgDelegateToGateway(
+		application.accAddress.String(),
+		gateway.accAddress.String(),
+	)
+
+	application.pendingMsgs = append(application.pendingMsgs, delegateMsg)
+}
+
+// sendStakeAndDelegateAppsTxs stakes the new applications and delegates them to both
+// the active and new gateways.
+// It also ensures that new gateways are delegated to the existing applications.
+func (s *relaysSuite) sendStakeAndDelegateAppsTxs(
+	sessionInfo *sessionInfoNotif,
+	newApps, newGateways []*accountInfo,
+) {
+
+	// TODO_IN_THIS_COMMIT: send an UpdateParams message to the application
```

Comment:
I think this TODO is old; we're currently doing this in `MoreActorsAreStakedAsFollows()`. It is necessary to set the `max_delegated_gateways` to the max number of gateways so that each application/gateway permutation may be exercised in each session. Otherwise, the relay batching logic becomes much more complex and our baseline expectations would change.